app/src/main/kotlin/compiler/syntax/AstFactory.kt

```diff
@@ -16,10 +16,29 @@ import compiler.regex.RegexDfa
 import compiler.syntax.LanguageGrammar.Productions
 import compiler.syntax.utils.TokenRegexParser
 
+typealias MutableStatementBlock = MutableList<Statement>
+typealias MutableGlobalBlock = MutableList<Program.Global>
+
 class AstFactory(private val diagnostics: Diagnostics) {
     class AstCreationFailed : CompilationFailed()
 
+    sealed class LowestAncestorBlock {
+        data class AncestorStatementBlock(val statementBlock: MutableStatementBlock, val previousStatement: Statement?) : LowestAncestorBlock()
+        data class AncestorGlobalBlock(val globalBlock: MutableGlobalBlock, val previousGlobal: Program.Global?) : LowestAncestorBlock()
+    }
+
+    private var auxiliaryGeneratorsCounter = 0
+    private fun getUniqueLabel(): String = "_pomocniczyPrzekaźnik${auxiliaryGeneratorsCounter++}"
```

Comment:
Starting with "_" should prevent any naming conflicts. Am I right?

Reply:
Underscores are allowed in identifiers in input source code, so there may be conflicts. You can use any character which is not normally allowed, and the name resolution shouldn't bother with this. You can also use get rid of the counter by surrounding the generator function by a dangling statement block (the same trick as in #178).

Reply:
(But considering how complicated the already written tests are, I won't insist on removing the counter.)